import_export/forms.py

```diff
@@ -91,3 +95,155 @@ class ExportForm(ImportExportFormBase):
     export_items = forms.MultipleChoiceField(
         widget=forms.MultipleHiddenInput(), required=False
     )
+
+
+class SelectableFieldsExportForm(ExportForm):
+    def __init__(self, formats, resources, *args, **kwargs):
+        super().__init__(formats, resources, *args, **kwargs)
+        self._init_selectable_fields(resources)
+
+    def _init_selectable_fields(self, resources: Iterable[ModelResource]) -> None:
+        """
+        Create `BooleanField(s)` for resource fields
+        """
+        self.resources = resources
+        self.is_selectable_fields_form = True
+        self.resource_fields = {resource.__name__: list() for resource in resources}
+
+        for resource in resources:
+            self._create_boolean_fields(resource)
+
+    def _create_boolean_fields(self, resource: ModelResource) -> None:
+        # Initiate resource to get ordered export fields
+        fields = resource().get_export_order()
+
+        boolean_fields = []  # will be used for ordering the fields
+
+        for field in fields:
+            field_name = self.create_boolean_field_name(resource, field)
+            boolean_field = forms.BooleanField(
+                label=field.replace("_", " ").title(),
+                initial=True,
+                required=False,
+            )
+            # These attributes will be used for rendering in template
+            boolean_field.is_selectable_field = True
+            boolean_field.resource_name = resource.__name__
+
+            self.fields[field_name] = boolean_field
+            boolean_fields.append(field_name)
+
+        self.order_fields(boolean_fields)  # Order fields by boolean fields
+
+        self.resource_fields[resource.__name__] = boolean_fields
+
+    @staticmethod
+    def create_boolean_field_name(resource: ModelResource, field_name: str) -> str:
+        """
+        Create field name by combining `resource_name` + `field_name` to prevent
+        confliction between resource fields with same name
+
+        Example:
+            BookResource            +   name -> bookresource_name
+            BookResourceWithNames   +   name -> bookresourcewithnames_name
+        """
+        return resource.__name__.lower() + "_" + field_name
+
+    def full_clean(self) -> None:
```

Comment:
Do we need this function?  Doesn't do anything?

Reply:
Definitely not needed, just a leftover